src/scripts/main.js

```diff
@@ -1,3 +1,261 @@
+  }
+};
+let clickedSortTable;
+
+tableHeaderCells.addEventListener('click', (e) => {
+  const headerChildren = [...tableHeaderCells.children];
+  const columnName = e.target.closest('th');
+
+  if (columnName !== null) {
+    const sortedColumn = headerChildren.indexOf(columnName);
+    const columnsToSort = [...tableBody.children];
+
+    if (clickedSortTable === columnName) {
+      isSecondClick = !isSecondClick;
+    } else {
+      isSecondClick = false;
+    }
+
+    clickedSortTable = columnName;
+
+    const columnsSorted = columnsToSort.sort((a, b) => {
+      const elemA = a.children[sortedColumn].innerHTML;
+      const elemB = b.children[sortedColumn].innerHTML;
+
+      const aToNum = toNumber(elemA);
+      const bToNum = toNumber(elemB);
+
+      if (Number.isNaN(aToNum)) {
+        return !isSecondClick
+          ? elemA.localeCompare(elemB)
+          : elemB.localeCompare(elemA);
+      }
+
+      return !isSecondClick ? aToNum - bToNum : bToNum - aToNum;
+    });
+
+    tableBody.innerHTML = '';
+
+    columnsSorted.forEach((el) => {
+      tableBody.appendChild(el);
+    });
+  }
+});
+
+tableBody.addEventListener('click', (e) => {
+  const headerChildren = [...tableBody.children];
+  const columnName = e.target.closest('tr');
+
+  headerChildren.forEach((el) => el.removeAttribute('class'));
+  columnName.classList.add('active');
+});
+
+const formContainer = document.createElement('form');
+const nameLabel = document.createElement('label');
+const nameInput = document.createElement('input');
+const posiitionLabel = document.createElement('label');
+const posiitionInput = document.createElement('input');
+const officeLabel = document.createElement('label');
+const officeChoose = document.createElement('select');
+const ageLabel = document.createElement('label');
+const ageInput = document.createElement('input');
+const salaryLabel = document.createElement('label');
+const salaryInput = document.createElement('input');
+const formButton = document.createElement('button');
+const cityOptions = [
+  `Tokyo`,
+  `Singapore`,
+  `London`,
+  `New York`,
+  `Edinburgh`,
+  `San Francisco`,
+];
+const inputs = [nameInput, posiitionInput, officeChoose, ageInput, salaryInput];
+
+const modifyElement = (parentEl, child, cellName, type) => {
+  parentEl.innerHTML = `${cellName}:`;
+  parentEl.setAttribute('data-qa', cellName.toLowerCase());
+  child.required = true;
+  child.type = type;
+  parentEl.appendChild(child);
+};
+
+modifyElement(nameLabel, nameInput, 'Name', 'text');
+modifyElement(posiitionLabel, posiitionInput, 'Position', 'text');
+modifyElement(ageLabel, ageInput, 'Age', 'number');
+modifyElement(salaryLabel, salaryInput, 'Salary', 'number');
+
+formContainer.classList.add('new-employee-form');
+
+officeLabel.innerHTML = 'Office:';
+officeLabel.setAttribute('data-qa', 'office');
+officeLabel.appendChild(officeChoose);
+officeChoose.required = true;
+formButton.innerText = 'Save to table';
```

Comment:
The button does not have a 'type' attribute. It's good practice to specify 'type="submit"' for buttons within forms.